#44 Travis CI integration & Jest for Unit Testing

JeremiahTee opened the pull request:
-added jest dependency for back-end, for unit testing purposes
     - so far it only has spec.js which will return True if jest correctly works
- integrated Travis CI by including a .travis.yml file
    - this yml file cd's into Server (back-end), runs npm install & test, then cd to Client (front-end) and runs npm install
    - finally, it spins up the docker containers by running docker-compose up and gracefully shuts down with docker-compose down

This resolves Issue #44

JeremiahTee committed:
add jest dependency, npm test passes

executing npm test will run spec.js inside src/ folder.
the test passes

JeremiahTee committed:
add travis.yml file for both back-end and front-end

JeremiahTee committed:
fix order of docker-compose in travis yml

no need to run docker-compose up in the front-end folder.

JeremiahTee committed:
test cd in travis yml

for some reason "cd .." does not work on Travis... build fails as the directories are wrong

JeremiahTee committed:
test out cd part 2

JeremiahTee committed:
remove unnecessary env variables & before-install step

before-install causes the build to fail because of the change directory.

let's put all the commands in the script, this should make all the builds pass (server, frontend and db)

JeremiahTee commented:
![image](https://user-images.githubusercontent.com/6520150/96360921-50937000-10ef-11eb-8d70-7f0d7f65a127.png)
![image](https://user-images.githubusercontent.com/6520150/96360931-630da980-10ef-11eb-9a52-6898e8a12b07.png)

GD-Dheer commented on .travis.yml:
```diff
@@ -0,0 +1,23 @@
+script:
+    - cd Server && npm install && npm test
+    - cd ../Client && npm install
+    - cd ../Server
```
Not necessary to go back into server to do a compose

JeremiahTee replied:
Sweet! I didn't know. I removed Line 19 
Solved with commit ea46e5d

GD-Dheer commented on .travis.yml:
```diff
@@ -0,0 +1,23 @@
+branches:
+  only:
+    - develop
+    - ci-integration-44
```
After some thought - we should run travis on develop & master

JeremiahTee replied:
Added master branch
Solved with commit ea46e5d

GD-Dheer commented:
<img width="1049" alt="Screen Shot 2020-10-18 at 3 32 21 PM" src="https://user-images.githubusercontent.com/25235293/96378008-41e0a380-1157-11eb-9d60-16746f110fc9.png">

Could you take a look? I triggered a new build on dev. @JeremiahTee

JeremiahTee committed:
resolve comments about travis in PR #62

https://github.com/Data-Boom/SOEN490/pull/62

PR comments:
- include master branch too
- remove Line 19 as no need to cd back into Server to spin up docker containers

JeremiahTee committed:
Merge branch 'develop' into ci-integration-44

Mehrdadmaskull reviewed (approved):
LGTM

JeremiahTee commented:
Looks like the failing build on develop is because Travis CI isn't properly set up there. As soon as this PR is approved, and ci-integration-44 branch is merged into develop, it will fix develop branch build and make it pass.

JeremiahTee committed:
remove ci-integration-44 branch from travis yml

as soon as the PR is merged, the branch will be deleted.
we only need develop and master on travis
